### PR TITLE
Document the sidecar.istio.io/statsInclusionPrefixes annotation

### PR DESCRIPTION
**Pull request by esnible**
Requested by @frankbu as a companion to https://github.com/istio/istio/pull/12353

### CODE REVIEW

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Envoy's statistics only cover the traffic for a particular Envoy instance.  See
+[Telemetry](/docs/tasks/telemetry/) for persistent per-service Istio telemetry.  The
+statistics recorded by Envoy can provide more information about specific pod instances.
+
+To see the statistics for a pod,
+
+```
+kubectl exec -it $POD  -c istio-proxy  -- sh -c 'curl localhost:15000/stats'
+```
```
Comment (venilnoronha) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```
{{< text bash >}}
$ kubectl exec -it $POD  -c istio-proxy  -- sh -c 'curl localhost:15000/stats'
{{< /text >}}
```

Use this?

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+See [the Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/cluster_manager/cluster_stats) for an explanation of the data recorded.
+
+By default, Istio configures Envoy to record minimal statistics.  The default collection
+keys are `cluster_manager`, `listener_manager`, `http_mixer_filter`, `tcp_mixer_filter`, `server`, and `cluster.xds-grpc`.  To see the Envoy settings for collection use  
+`istioctl proxy-config bootstrap` following [these instructions](https://istio.io/help/ops/traffic-management/proxy-cmd/).
+Consult the `statsMatcher` JSON element.
+
+Configure Envoy to record statistics for inbound or outbound traffic by adding the
+`sidecar.istio.io/statsInclusionPrefixes` annotation to the pod template in a Kubernetes Deployment.
```
Comment (venilnoronha) [inline content/help/ops/telemetry/envoy-stats/index.md]: nit: s/Deployment/\`Deployment\`

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+Add `cluster.outbound` for gather data about outbound traffic activity and circuit breaking.
+To gather data on inbound traffic add `listener`.  A sample annotation including `cluster.outbound`
```
Comment (venilnoronha) [inline content/help/ops/telemetry/envoy-stats/index.md]: nit: s/traffic/traffic,

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+can be seen in _samples/httpbin/sample-client/fortio-deploy.yaml_.
```
Comment (venilnoronha) [inline content/help/ops/telemetry/envoy-stats/index.md]: Use a real link to the file instead?

Comment (esnible) [reply content/help/ops/telemetry/envoy-stats/index.md]: Do you suggest a raw link to github master branch, like https://raw.githubusercontent.com/istio/istio/master/samples/httpbin/sample-client/fortio-deploy.yaml ?

Comment (venilnoronha) [reply content/help/ops/telemetry/envoy-stats/index.md]: Never mind, leave it as-is. I think other docs just indent the text as you have.

Comment (frankbu) [reply content/help/ops/telemetry/envoy-stats/index.md]: There are a bunch of guidelines for contributing to the docs that you should make sure to follow:

https://deploy-preview-3581--preliminary-istio.netlify.com/about/contribute/creating-and-editing-pages/#linking-to-other-pages

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+weight: 95
+---
+
+The Envoy proxy keeps detailed statistics about network traffic.
+
+Envoy's statistics only cover the traffic for a particular Envoy instance.  See
+[Telemetry](/docs/tasks/telemetry/) for persistent per-service Istio telemetry.  The
+statistics recorded by Envoy can provide more information about specific pod instances.
+
+To see the statistics for a pod,
```
Comment (frankbu) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
To see the statistics for a pod:
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+{{< text bash >}}
+$ kubectl exec -it $POD  -c istio-proxy  -- sh -c 'curl localhost:15000/stats'
+{{< /text >}}
+
+See [the Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/cluster_manager/cluster_stats) for an explanation of the data recorded.
+
+By default, Istio configures Envoy to record minimal statistics.  The default collection
+keys are `cluster_manager`, `listener_manager`, `http_mixer_filter`, `tcp_mixer_filter`, `server`, and `cluster.xds-grpc`.  To see the Envoy settings for collection use
```
Comment (frankbu) [inline content/help/ops/telemetry/envoy-stats/index.md]: for the collection

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+title: Envoy Statistics
+description: Fine-grained control of Envoy statistics.
+weight: 95
+---
+
+The Envoy proxy keeps detailed statistics about network traffic.
+
+Envoy's statistics only cover the traffic for a particular Envoy instance.  See
+[Telemetry](/docs/tasks/telemetry/) for persistent per-service Istio telemetry.  The
+statistics recorded by Envoy can provide more information about specific pod instances.
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
statistics the envoy proxies record can provide more information about specific pod instances.
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+To see the statistics for a pod:
+
+{{< text bash >}}
+$ kubectl exec -it $POD  -c istio-proxy  -- sh -c 'curl localhost:15000/stats'
+{{< /text >}}
+
+See [the Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/cluster_manager/cluster_stats) for an explanation of the data recorded.
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
See [the Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/cluster_manager/cluster_stats)
for an explanation of the data recorded.
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+By default, Istio configures Envoy to record minimal statistics.  The default collection
+keys are `cluster_manager`, `listener_manager`, `http_mixer_filter`, `tcp_mixer_filter`, `server`, and `cluster.xds-grpc`.  To see the Envoy settings for statistics data collection use
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
keys are:

- `cluster_manager`
- `listener_manager`
- `http_mixer_filter`
- `tcp_mixer_filter`
- `server`
- `cluster.xds-grpc`

To see the Envoy settings for statistics data collection use
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+`istioctl proxy-config bootstrap` following [these instructions](https://istio.io/help/ops/traffic-management/proxy-cmd/).
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
`istioctl proxy-config bootstrap` and follow the [deep dive into Envoy configuration](/help/ops/traffic-management/proxy-cmd/#deep-dive-into-envoy-configuration).
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+Consult the `statsMatcher` JSON element.
+
+Configure Envoy to record statistics for inbound or outbound traffic by adding the
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
To configure Envoy to record statistics for inbound or outbound traffic, add the
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+`sidecar.istio.io/statsInclusionPrefixes` annotation to the pod template in a Kubernetes `Deployment`.
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
`sidecar.istio.io/statsInclusionPrefixes` annotation to the pod template in the Kubernetes deployment.
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+Add `cluster.outbound` for gather data about outbound traffic activity and circuit breaking.
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
Add the `cluster.outbound` annotation to gather data about outbound traffic activity and circuit breaking.
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+{{< text bash >}}
+$ kubectl exec -it $POD  -c istio-proxy  -- sh -c 'curl localhost:15000/stats'
+{{< /text >}}
+
+See [the Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/cluster_manager/cluster_stats) for an explanation of the data recorded.
+
+By default, Istio configures Envoy to record minimal statistics.  The default collection
+keys are `cluster_manager`, `listener_manager`, `http_mixer_filter`, `tcp_mixer_filter`, `server`, and `cluster.xds-grpc`.  To see the Envoy settings for statistics data collection use
+`istioctl proxy-config bootstrap` following [these instructions](https://istio.io/help/ops/traffic-management/proxy-cmd/).
+Consult the `statsMatcher` JSON element.
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]: This line lacks context. When would a reader consult the element?

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+Configure Envoy to record statistics for inbound or outbound traffic by adding the
+`sidecar.istio.io/statsInclusionPrefixes` annotation to the pod template in a Kubernetes `Deployment`.
+Add `cluster.outbound` for gather data about outbound traffic activity and circuit breaking.
+To gather data on inbound traffic, add `listener`.  A sample annotation including `cluster.outbound`
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
To gather data on inbound traffic, add the `listener` annotation.  The following sample annotation includes `cluster.outbound`
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+can be seen in `samples/httpbin/sample-client/fortio-deploy.yaml`.
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion

{{< text_file file="/samples/httpbin/sample-client/fortio-deploy.yaml" syntax="yaml">}}
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+
+You can cause Envoy to gather less data than usual by overriding the defaults.  Use
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
You can override the Envoy defaults to gather less data than usual.  Use the
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+`sidecar.istio.io/statsInclusionPrefixes: cluster_manager,listener_manager` for the least possible statistics
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
`sidecar.istio.io/statsInclusionPrefixes: cluster_manager,listener_manager` key/value pair
to collect the least statistics possible
```

File: content/help/ops/telemetry/envoy-stats/index.md [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+collection.
```
Comment (rcaballeromx) [inline content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
```

File: content/help/ops/telemetry/envoy-stats/index.md (hunk 16 above)
Comment (esnible) [reply content/help/ops/telemetry/envoy-stats/index.md]: This suggestion gives me ```
Error: Error building site: "/site/content/help/ops/telemetry/envoy-stats/index.md:42:1": failed to render shortcode "text_file": failed to process shortcode: "/site/layouts/shortcodes/text_file.html:7:13": execute of template failed: template: shortcodes/text_file.html:7:13: executing "shortcodes/text_file.html" at <readFile (.Get "file...>: error calling readFile: stat /site/content_zh/samples/httpbin/sample-client/fortio-deploy.yaml: no such file or directory
```  I will try `<github_file>` instead.

Comment (frankbu) [reply content/help/ops/telemetry/envoy-stats/index.md]:
```suggestion
can be seen in [fortio-deploy.yaml]({{< github_file >}}/samples/httpbin/sample-client/fortio-deploy.yaml`).
```

**Pull request merged**